tests/unit/models/test_model_interfaces.py

```diff
@@ -406,27 +417,62 @@ def test_find_best_model_initialization_only_changes_params_with_priors(
     model.find_best_model_initialization(2)
 
     npt.assert_allclose(1.0, model.model.kernel.variance)
+    npt.assert_array_equal(dim, model.model.kernel.lengthscales.shape)
+    npt.assert_raises(
+        AssertionError, npt.assert_allclose, [0.2, 0.2], model.model.kernel.lengthscales
+    )
+
+
+@pytest.mark.parametrize("dim", [1, 10])
+def test_find_best_model_initialization_changes_params_with_sigmoid_bjectors(
+    gpr_interface_factory, dim: int
+) -> None:
+    x = tf.constant(np.arange(1, 5).reshape(-1, 1), dtype=gpflow.default_float())  # shape: [4, 1]
+    model = gpr_interface_factory(x, _3x_plus_10(x))
+    model.model.kernel = gpflow.kernels.RBF(lengthscales=[0.2] * dim)
+
+    if isinstance(model, (VariationalGaussianProcess, SparseVariational)):
+        pytest.skip("find_best_model_initialization is only implemented for the GPR models.")
+
+    upper = tf.cast([10.0] * dim, dtype=tf.float64)
+    lower = upper / 100
+    model.model.kernel.lengthscales = gpflow.Parameter(
```

Comment:
Have you tested parameters with both constraints _and_ priors?

Reply:
Gpflow allows it. I will add the test.